src/box/box.cc

```diff
@@ -1151,24 +1151,49 @@ box_check_auth_type(void)
 }
 
 static enum election_mode
-box_check_election_mode(void)
+election_mode_by_name(const char *name)
 {
-	const char *mode = cfg_gets("election_mode");
-	if (strcmp(mode, "off") == 0)
+	if (strcmp(name, "off") == 0)
 		return ELECTION_MODE_OFF;
-	else if (strcmp(mode, "voter") == 0)
+	else if (strcmp(name, "voter") == 0)
 		return ELECTION_MODE_VOTER;
-	else if (strcmp(mode, "manual") == 0)
+	else if (strcmp(name, "manual") == 0)
 		return ELECTION_MODE_MANUAL;
-	else if (strcmp(mode, "candidate") == 0)
+	else if (strcmp(name, "candidate") == 0)
 		return ELECTION_MODE_CANDIDATE;
 
 	diag_set(ClientError, ER_CFG, "election_mode",
-		"the value must be one of the following strings: "
-		"'off', 'voter', 'candidate', 'manual'");
+		  "the value must be one of the following strings: "
+		  "'off', 'voter', 'candidate', 'manual'");
 	return ELECTION_MODE_INVALID;
 }
 
+static enum election_mode
+election_mode_by_name_xc(const char *name)
```

Comment:
Nit: since the function is used only in one place, we can get rid of it and explicitly raise the diag in `box_check_replication_anon`

Reply:
Fixed.